components/proxy-service/internal/authorization/oauth/oauthclient.go

```diff
@@ -22,53 +22,58 @@ type oauthResponse struct {
 	Scope       string `json:"scope"`
 }
 
-type OAuthClient interface {
+type Client interface {
 	GetToken(clientID string, clientSecret string, authURL string) (string, apperrors.AppError)
 	InvalidateAndRetry(clientID string, clientSecret string, authURL string) (string, apperrors.AppError)
+	InvalidateTokenCache(clientID string)
 }
 
-type oauthClient struct {
+type client struct {
 	timeoutDuration int
 	tokenCache      tokencache.TokenCache
 }
 
-func NewOauthClient(timeoutDuration int, tokenCache tokencache.TokenCache) OAuthClient {
-	return &oauthClient{
+func NewOauthClient(timeoutDuration int, tokenCache tokencache.TokenCache) Client {
+	return &client{
 		timeoutDuration: timeoutDuration,
 		tokenCache:      tokenCache,
 	}
 }
 
-func (oc *oauthClient) GetToken(clientID string, clientSecret string, authURL string) (string, apperrors.AppError) {
-	token, found := oc.tokenCache.Get(clientID)
+func (c *client) GetToken(clientID string, clientSecret string, authURL string) (string, apperrors.AppError) {
+	token, found := c.tokenCache.Get(clientID)
 	if found {
-		return "Bearer " + token, nil
+		return token, nil
 	}
 
-	tokenResponse, err := oc.requestToken(clientID, clientSecret, authURL)
+	tokenResponse, err := c.requestToken(clientID, clientSecret, authURL)
 	if err != nil {
 		return "", err
 	}
 
-	oc.tokenCache.Add(clientID, tokenResponse.AccessToken, tokenResponse.ExpiresIn)
+	c.tokenCache.Add(clientID, tokenResponse.AccessToken, tokenResponse.ExpiresIn)
 
-	return "Bearer " + tokenResponse.AccessToken, nil
+	return tokenResponse.AccessToken, nil
 }
 
-func (oc *oauthClient) InvalidateAndRetry(clientID string, clientSecret string, authURL string) (string, apperrors.AppError) {
-	oc.tokenCache.Remove(clientID)
+func (c *client) InvalidateAndRetry(clientID string, clientSecret string, authURL string) (string, apperrors.AppError) {
+	c.tokenCache.Remove(clientID)
 
-	tokenResponse, err := oc.requestToken(clientID, clientSecret, authURL)
+	tokenResponse, err := c.requestToken(clientID, clientSecret, authURL)
 	if err != nil {
 		return "", err
 	}
 
-	oc.tokenCache.Add(clientID, tokenResponse.AccessToken, tokenResponse.ExpiresIn)
+	c.tokenCache.Add(clientID, tokenResponse.AccessToken, tokenResponse.ExpiresIn)
 
-	return "Bearer " + tokenResponse.AccessToken, nil
+	return tokenResponse.AccessToken, nil
 }
 
-func (oc *oauthClient) requestToken(clientID string, clientSecret string, authURL string) (*oauthResponse, apperrors.AppError) {
+func (c *client) InvalidateTokenCache(clientID string) {
+	c.tokenCache.Remove(clientID)
+}
+
+func (c *client) requestToken(clientID string, clientSecret string, authURL string) (*oauthResponse, apperrors.AppError) {
 	transport := &http.Transport{
 		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Is it a good idea to keep InsecureSkipVerify always set to true? In my opinion this should be passed as a paremeter / env variable but that's rather idea for a follow-up task. What do you think on this?

Reply:
There is 'skipVerify' flag used when creating ReverseProxy. I think we could use the same flag and pass it to oauthclient.

Reply:
The following issue has been added : https://github.com/kyma-project/kyma/issues/1639